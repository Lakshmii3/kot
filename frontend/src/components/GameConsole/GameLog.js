import React from 'react';

import { List } from 'antd';
import "./GameLog.css";

const GameLog = (props) => {
    return (
        <div>
            <List
                // header={<div>Header</div>}
                // footer={<div>Footer</div>}
                bordered
                dataSource={props.data}
                renderItem={item => (
                    <List.Item>
                        {item}
                    </List.Item>
                )}
            />
        </div>
    );
}


export default GameLog;