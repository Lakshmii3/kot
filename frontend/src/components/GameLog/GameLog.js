import React, { Component } from 'react';

import "./GameLog.css";

export default class GameLog extends Component {
  render() {
    return (
      <div className="gamelog">
        { this.props.children }
      </div>
    )
  }
}
