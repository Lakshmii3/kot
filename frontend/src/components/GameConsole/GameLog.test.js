import React from 'react';
import { shallow } from 'enzyme';

import GameLog from './GameLog';

describe('Verify GameLog layout template', () => {
    it('GameLog should render correctly', () => {
        const errorPage = shallow(<GameLog />);
        expect(errorPage).toMatchSnapshot();
    });

    it('Verify parameter provided logs are rendered correctly', () => {
        // class FaceValues(enum):
        // One = 0
        // Two = 1
        // Three = 2
        // Attack = 3
        // Heart = 4
        // Energy = 5

        const tLog = [
            '1',
            '2',
            '3',
            'attack',
            'heal',
            'energy',];

        const component = shallow(<Gamelog data={tLog} />);
        expect(component).toMatchSnapshot();
    });
})