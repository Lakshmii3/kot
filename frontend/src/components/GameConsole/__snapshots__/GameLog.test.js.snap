// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verify GameLog layout template GameLog should render correctly 1`] = `
<div>
  <List
    bordered={true}
    dataSource={Array []}
    loading={false}
    pagination={false}
    renderItem={[Function]}
    split={true}
  />
</div>
`;

exports[`Verify GameLog layout template Verify parameter provided logs are rendered correctly 1`] = `
<div>
  <List
    bordered={true}
    dataSource={
      Array [
        "1",
        "2",
        "3",
        "attack",
        "heal",
        "energy",
      ]
    }
    loading={false}
    pagination={false}
    renderItem={[Function]}
    split={true}
  />
</div>
`;
